In [1]:
import pandas as pd
import numpy as np
import math

In [4]:
df = pd.read_csv(r"C:\Users\naira\OneDrive\Documents\property.csv")

In [5]:
df.head()

Suburb           Address  Rooms Type    Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000     VB  Nelson   

         Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  03-12-2016       2.5      3067  ...         1  1.0       202           NaN   
1  04-02-2016       2.5      3067  ...         1  0.0       156          79.0   
2  04-03-2017       2.5      3067  ...         2  0.0       134         150.0   
3  04-03-2017       2.5      3067  ...         2  1.0        94           NaN   
4  04-06-2016       2.5      3067  ...         1  2.0       120         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0          4019  
1          4019  
2          4019  
3          4019  
4          4019  

[5 rows x 21 columns]

## Q1: Hypothesis Test – Altona

Null: Mean price in Altona = 800000  
Alternate1: Mean price in Altona > 80000


In [6]:
altona_prices = df[df['Suburb'] == 'Altona']['Price'].dropna()

In [7]:
t_stat = (altona_prices.mean() - 800000) / (
    altona_prices.std(ddof=1) / math.sqrt(len(altona_prices))
)
t_stat

np.float64(1.0277020770199676)

AN: The calculated t value is 1.03, which is less than the critical value of 1.645.
Therefore, the null hypothesis is not rejected, and there is insufficient statistical evidence to conclude that the average property price in Altona is greater than $800,000.

## Q2. Summer vs Winter Property Prices (2016)

Null: summer = winter

alternate: summer not equal to winter

In [19]:
df['Date'] = pd.to_datetime(df['Date'], dayfirst=True)
df_2016 = df.loc[df['Date'].dt.year == 2016].copy()
df_2016['Month'] = df_2016['Date'].dt.month

In [18]:
winter = df_2016[df_2016['Month'].isin([10, 11, 12, 1, 2, 3])]['Price'].dropna()
summer = df_2016[df_2016['Month'].isin([4, 5, 6, 7, 8, 9])]['Price'].dropna()

In [11]:
t_stat = (winter.mean() - summer.mean()) / np.sqrt(
    winter.var(ddof=1)/len(winter) +
    summer.var(ddof=1)/len(summer)
)
t_stat

np.float64(3.9211012778003753)

AN: An independent two-sample t-test was conducted at the 5% significance level to compare property prices sold during summer and winter months in 2016.
The calculated t-statistic is 3.92, which exceeds the critical value of > 1.96.
Therefore, the null hypothesis is rejected, and there is a statistically significant difference between summer and winter property prices in 2016.

## Q3. Probability Question – Abbotsford (Car Parking)

In [17]:
abford = df[df['Suburb'] == 'Abbotsford']

In [16]:
no_cp = (abford['Car'] == 0).mean()
round(no_cp,3)

np.float64(0.268)

AN: The probability that exactly 3 out of 10 properties sold in Abbotsford do not have a car parking space is 0.260

## Q4: Probability of a 3-Room Property in Abbotsford

In [25]:
prob3_rooms = (abford['Rooms'] == 3).mean()
round(prob3_rooms, 3)

np.float64(0.357)

AN: There is a 35.7% chance that a randomly chosen property in Abbotsford has three rooms.

## Q5. Probability of a 2-Bathroom Property in Abbotsford

In [26]:
prob2_bath = (abford['Bathroom'] == 2).mean()
round(prob2_bath, 3)

np.float64(0.339)

AN: There is a 33.9% chance that a property chosen at random from Abbotsford has two bathrooms.

## Q6: Hypothesis Test – Richmond

Null - = 1000000

alternate not equal to 1000000

In [28]:
richmond_prices = df[df['Suburb'] == 'Richmond']['Price'].dropna()

In [30]:
t_stat = (richmond_prices.mean() - 1_000_000) / (
    richmond_prices.std(ddof=1) / math.sqrt(len(richmond_prices))
)
round(t_stat,3)

np.float64(2.58)

p-value
p-value = 0.010

AN: The calculated t-value is 2.58, which is above the critical values of 1.96, therefore, the null hypothesis is rejected.

## Q7: Therefore, the null hypothesis is rejected.

choice of test
2 independent group
- properties with car parking
- properties without car parking

comparing aveage prices

population SD is unknown

null = with cp = without cp

alternate = with cp > without cp

In [33]:
with_car = df[df['Car'] > 0]['Price'].dropna()
without_car = df[df['Car'] == 0]['Price'].dropna()

In [35]:
t_stat = (with_car.mean() - without_car.mean()) / math.sqrt(
    with_car.var(ddof=1)/len(with_car) +
    without_car.var(ddof=1)/len(without_car)
)
round(t_stat,3)

np.float64(-0.272)

p-value = 0.785

From a business perspective, this result suggests that providing car parking alone does not guarantee a higher selling price.
Developers should consider other factors as well.

## Q8: Property are influenced by

In training ANOVA was not covered

## Q9. p-Value Interpretation (Decision Making)

P value is 0.032
The observed difference is unlikely to be due to chance alone

The evidence contradicts the null hypothesis

Since the p-value is less than 0.05 hypothesis the decision is to reject the null hypothesis.

From a business perspective, this result suggests that there is a statistically significant difference in property prices between the two suburbs.
This means that location likely plays an important role in pricing, and the observed difference is unlikely to be due to random variation alone.

## Q10. Industry-Style Hypothesis Validation

identify the correct test state hypothesis • Report p-value • Give a clear recommendation to policymakers

Identify the correct test = right tailed test


null = average price of 2 bathrooms = average price with 2 or few bathrooms

alternate = average price of 2 bathrooms > average price with 2 or few bathrooms

p-value = 0.001

reject the null hypothesis since p-value < 0.05

The statistical analysis provides strong evidence that properties with more than 2 bathrooms are a significantly higher average price.
Policymakers and housing planners should therefore recognize additional bathrooms as a meaningful value driver.